src/main/java/com/nekokittygames/thaumictinkerer/common/blocks/BlockEnchantmentPillar.java

```diff
@@ -25,104 +26,104 @@
-            state = state.withProperty(Top, true);
-            curMeta -= 8;
-        } else {
-            state = state.withProperty(Top, false);
-        }
-        return state.withProperty(Direction, curMeta);
-    }
+  @Override
+  public int getMetaFromState(IBlockState state) {
+    return state.getValue(Direction) + (state.getValue(Top) ? 8 : 0);
+  }
 
-    @Override
-    protected boolean isInCreativeTab() {
-        return false;
+  @Override
+  public IBlockState getStateFromMeta(int meta) {
+    int curMeta = meta;
+    IBlockState state = this.blockState.getBaseState();
+    if (curMeta >= 8) {
+      state = state.withProperty(Top, true);
+      curMeta -= 8;
+    } else {
+      state = state.withProperty(Top, false);
     }
+    return state.withProperty(Direction, curMeta);
+  }
 
-    public EnumBlockRenderType getRenderType(IBlockState state) {
-        if (!state.getValue(Top))
-            return EnumBlockRenderType.MODEL;
-        else
-            return EnumBlockRenderType.INVISIBLE;
-    }
+  @Override
+  protected boolean isInCreativeTab() {
+    return false;
+  }
 
-    private BlockPos IsEnchanterPos(World world,BlockPos pos)
-    {
-        if(world.getBlockState(pos).getBlock()==ModBlocks.osmotic_enchanter)
-            return pos;
-        if(world.getBlockState(pos.down(1)).getBlock()==ModBlocks.osmotic_enchanter)
-            return pos.down(1);
-        return null;
-    }
-    @Override
-    public void breakBlock(World worldIn, BlockPos pos, IBlockState state) {
-        BlockPos enchanterLoc=null;
-        enchanterLoc=IsEnchanterPos(worldIn,pos.south(3));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn,pos.west(3));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn,pos.north(3));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn,pos.east(3));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn, pos.south(2).west(2));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn, pos.north(2).west(2));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn, pos.south(2).east(2));
-        if(enchanterLoc==null)
-            enchanterLoc = IsEnchanterPos(worldIn, pos.north(2).east(2));
+  public EnumBlockRenderType getRenderType(IBlockState state) {
+    if (!state.getValue(Top))
+      return EnumBlockRenderType.MODEL;
+    else
+      return EnumBlockRenderType.INVISIBLE;
+  }
 
-        ThaumicTinkerer.logger.info("Enchanter Pos Found - "+ (enchanterLoc != null ? enchanterLoc.toString() : "null"));
-        if(enchanterLoc!=null)
-        {
-            TileEntity te=worldIn.getTileEntity(enchanterLoc);
-            if(te instanceof TileEntityEnchanter)
-            {
-                ((TileEntityEnchanter)te).BreakPillars();
-            }
-        }
+  private BlockPos IsEnchanterPos(World world, BlockPos pos) {

```

Comment:
Codacy found an issue: [The instance method name 'IsEnchanterPos' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/nekosune/ThaumicTinkerer/pullRequest?prid=6411795)